src/main/java/com/ibm/g11n/pipeline/client/ServiceClient.java

```diff
@@ -441,6 +442,111 @@ public abstract void updateResourceEntry(String bundleId, String language,
             String resKey, ResourceEntryDataChangeSet changeSet)
                     throws ServiceException;
 
+    //
+    // {serviceInstanceId}/v2/documents APIs
+    //
+
+    /**
+     * Returns a set of document IDs available in the service instance.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param type The type of documents being requested.
+     * @return A set of document IDs.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract Set<String> getDocumentIds(DocumentType type) throws ServiceException;
+
+    /**
+     * Creates a new translatble document.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId
+     *          The new document's ID. Mut not be null or empty.
+     *          <br>The document ID must match a regular expression pattern
+     *          [a-zA-Z0-9][a-zA-Z0-9_.-]* and the length must be less than or equal
+     *          to 255.
+     * @param type The type of document being created.
+     * @param newDocumentData
+     *          The new document's configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void createDocument(String documentId, DocumentType type,
```

Comment:
I propose to swap the order of parameter - DocumentType type first, then String documentId.   A document ID is unique within a same type - so DocumentType should be upper level concept. This applies to all other doc APIs in this class.